### DISCUSSION TITLE
SSL rabbit hole and NextCloud installation

### DISCUSSION THREAD

**Discussion by udp1024** [nextcloud/all-in-one#4664] [Ideas] [answered]
Reading through the documentation and figuring out how I want to install the NextCloud app in a docker container to try, it feels as if the project has fallen into a rabbit-hole of trying to automate getting an SSL certificate.

Why isn't there an option to provide a valid certificate and a key file either through volume mount or by copying via docker compose?

For someone like me, I DO NOT wish to expose NextCloud to the Internet in any shape or form. Yet, I still use valid SSL certificates, either from LetsEncrypt or my private CA that is also running in the sandbox. All access to my network is vai VPN or not at all.

I am sure there was a lot of consideration and though that went into the application design, but this SSL/TLS imbroglio is holding it back. Just my $0.02. But I feel there is a valid use case for providing a path to use a pre-provisioned SSL Certificate and key.

**Comment by szaimen** [comment, marked as answer]
Hi, see https://github.com/nextcloud/all-in-one/tree/main?tab=readme-ov-file#are-self-signed-certificates-supported-for-nextcloud

Also see https://github.com/nextcloud/all-in-one/discussions/588#discussioncomment-3450343 but it is not supported.
